scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1025,24 +1025,28 @@
     (when (seq running-tasks)
       (log/info "Preparing to reconcile" (count running-tasks) "tasks")
       ;; TODO: When turning on periodic reconcilation, probably want to move this to startup
-      (doseq [[task-id] running-tasks
-              :let [task-ent (d/entity db [:instance/task-id task-id])
-                    hostname (:instance/hostname task-ent)
-                    job (util/job-ent->map (:job/_instance task-ent))
-                    task-request (make-task-request db job :task-id task-id)]]
-        ;; Need to lock on fenzo when accessing taskAssigner because taskAssigner and
-        ;; scheduleOnce can not be called at the same time.
-        (locking fenzo
-          (.. fenzo
-              (getTaskAssigner)
-              (call task-request hostname))))
-      (doseq [ts (partition-all 50 running-tasks)]
-        (log/info "Reconciling" (count ts) "tasks, including task" (first ts))
-        (mesos/reconcile-tasks driver (mapv (fn [[task-id status slave-id]]
-                                              {:task-id {:value task-id}
-                                               :state (sched->mesos status)
-                                               :slave-id {:value slave-id}})
-                                            ts)))
+      (let [processed-tasks (->> (for [[task-id] running-tasks
+                                       :let [task-ent (d/entity db [:instance/task-id task-id])
+                                             hostname (:instance/hostname task-ent)]]
+                                   (when-let [job (util/job-ent->map (:job/_instance task-ent))]
+                                     (let [task-request (make-task-request db job :task-id task-id)]
+                                       ;; Need to lock on fenzo when accessing taskAssigner because taskAssigner and
+                                       ;; scheduleOnce can not be called at the same time.
+                                       (locking fenzo
+                                         (.. fenzo
+                                             (getTaskAssigner)
+                                             (call task-request hostname)))
+                                       [task-id])))
+                                 (remove nil?))]
+        (when (not= (count running-tasks) (count processed-tasks))
+          (log/info "Skipping reconciling" (- (count running-tasks) (count processed-tasks)) "tasks"))
```

Comment:
Maybe this should be `log/error`?

Reply:
We already logged the error in `job-ent->map`

Reply:
Wouldn't we want this line to show up in the ERROR-only log?

Reply:
@pschorf tie-breaker?

Reply:
I'd say it's an error

Reply:
Done.